src/jit/lsra.cpp

```diff
@@ -6709,6 +6785,13 @@ void LinearScan::processBlockStartLocations(BasicBlock* currentBlock, bool alloc
                     }
                     inVarToRegMap[assignedInterval->getVarIndex(compiler)] = REG_STK;
                 }
+#ifdef _TARGET_ARM_
+                // Consider overlapping floating point register for TYP_DOUBLE
+                else if (!assignedInterval->isConstant && isOverlappedRegRecord(assignedInterval, physRegRecord))
```

Comment:
>isOverlappedRegRecord(assignedInterval, physRegRecord) [](start = 58, length = 54)

It appears that this will return true for the case where `assignedInterval` is `TYP_FLOAT` but its `assignedReg` is not `physRegRecord` (i.e. the case below). In this case this assert will fire incorrectly.